server/mscalendar/subscription.go

```diff
@@ -115,15 +118,16 @@ func (m *mscalendar) DeleteMyEventSubscription() error {
 
 	subscriptionID := m.actingUser.Settings.EventSubscriptionID
 
-	err = m.Store.DeleteUserSubscription(m.actingUser.User, subscriptionID)
+	err = m.DeleteOrphanedSubscription(subscriptionID)
 	if err != nil {
-		return errors.WithMessagef(err, "failed to delete subscription %s", subscriptionID)
+		return err
 	}
 
-	err = m.DeleteOrphanedSubscription(subscriptionID)
+	err = m.Store.DeleteUserSubscription(m.actingUser.User, subscriptionID)
 	if err != nil {
-		return err
+		return errors.WithMessagef(err, "failed to delete subscription %s", subscriptionID)
 	}
+
```

Comment:
Logic makes sense to me. Remote remote first, then local right?

Reply:
Yes, try to remove the remote first, and if it fails, do not remove the local. I got my test environment in a state where I failed to remove some remote subscription but the local data was removed previously, so I could track anything back.